app/models/concerns/operations/stock_operations.rb

```diff
@@ -68,5 +70,98 @@ def register_loss(package, quantity:, from_location:)
         location: from_location
       )
     end
+
+    def pack_or_unpack(params, user_id)
+      raise ActionNotAllowedError.new unless PACK_UNPACK_ALLOWED_ACTIONS.include?(params[:task])
+      PackUnpack.new(params, user_id).public_send(params[:task])
```

Comment:
@abhinav098 here not able to locate the definition of `public_send`.. can you please point it?

Reply:
@swatijadhav , it's a ruby method. https://apidock.com/ruby/Object/public_send
Public send calls the public functions of a class.